pkgs/by-name/de/deno/package.nix

```diff
@@ -57,28 +77,88 @@ rustPlatform.buildRustPackage rec {
   # To avoid this we pre-download the file and export it via RUSTY_V8_ARCHIVE
   env.RUSTY_V8_ARCHIVE = librusty_v8;
 
-  # Tests have some inconsistencies between runs with output integration tests
-  # Skipping until resolved
-  doCheck = false;
+  preCheck =
+    ''
+      # Integration tests need a writable home dir
+      export HOME=$(mktemp -d)
+    ''
+    + lib.optionalString stdenv.hostPlatform.isDarwin ''
+      # Unset the env var defined by bintools-wrapper because it triggers Deno's sandbox protection in some tests.
+      # ref: https://github.com/denoland/deno/pull/25271
+      unset LD_DYLD_PATH
+    '';
+
+  cargoTestFlags = [
+    "--lib" # unit tests
+    "--test integration_tests"
+    # Test targets not included here:
+    # - node_compat: there are tons of network access in them and it's not trivial to skip test cases.
+    # - specs: this target uses a custom test harness that doesn't implement the --skip flag.
+    #   refs:
+    #   - https://github.com/denoland/deno/blob/2212d7d814914e43f43dfd945ee24197f50fa6fa/tests/Cargo.toml#L25
+    #   - https://github.com/denoland/file_test_runner/blob/9c78319a4e4c6180dde0e9e6c2751017176e65c9/src/collection/mod.rs#L49
+  ];
+  checkFlags =
+    [
+      # Internet access
+      "--skip=check::ts_no_recheck_on_redirect"
+      "--skip=js_unit_tests::quic_test"
+      "--skip=node_unit_tests::http_test"
+      "--skip=node_unit_tests::http2_test"
+      "--skip=node_unit_tests::net_test"
+      "--skip=node_unit_tests::tls_test"
+
+      # GPU access
+      "--skip=js_unit_tests::webgpu_test"
+      "--skip=js_unit_tests::jupyter_test"
+
+      # Use of /usr/bin
+      "--skip=specs::permission::proc_self_fd"
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      # Test hangs, needs investigation
+      "--skip=repl::pty_complete_imports_no_panic_empty_specifier"
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      # Expects specific shared libraries from macOS to be linked
+      "--skip=shared_library_tests::macos_shared_libraries"
+
+      # Darwin sandbox issues
+      "--skip=watcher"
+      "--skip=node_unit_tests::_fs_watch_test"
+      "--skip=js_unit_tests::fs_events_test"
+    ];
+
+  __darwinAllowLocalNetworking = true;
+
+  nativeCheckInputs = [
+    curl
+    nodejs
+    git
+    python3
+  ];
 
   preInstall = ''
-    find ./target -name libswc_common${stdenv.hostPlatform.extensions.sharedLibrary} -delete
+    # Delete generated shared libraries that aren't needed in the final package
+    find ./target \
+      -name "libswc_common${stdenv.hostPlatform.extensions.sharedLibrary}" -o \
+      -name "libtest_ffi${stdenv.hostPlatform.extensions.sharedLibrary}" -o \
+      -name "libtest_napi${stdenv.hostPlatform.extensions.sharedLibrary}" \
+      -delete
   '';
 
   postInstall = lib.optionalString canExecute ''
+    rm $out/bin/denort
+    rm $out/bin/test_server
```

Comment:
This needs to live outside of `canExecute` please or cross-platform builds will end up with additional binaries.

---

Also this method is opt-out of extra bits, but do we want to be opt-in instead? It'd just mean we wouldn't accidentally package more testing utils etc without noticing. This part isn't a big blocker.